### PR TITLE
Use demo values for demo

### PR DESCRIPTION
**Pull request by howardjohn**
Hopefully fix https://github.com/istio/installer/issues/125

Most of changes are from updating the demo yaml, the real change is the values.yaml

### CODE REVIEW

File: test/demo/k8s.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@ metadata:
...
..."requests":{"cpu":"10m","memory":"40Mi"}},"statusPort":15020,"tracer":"zipkin"},"proxy_init":{"image":"proxy_init"},"sds":{"enabled":false,"udsPath":"","useNormalJwt":false,"useTrustworthyJwt":false},"sidecarInjectorWebhook":{"enabled":true,"rewriteAppHTTPProbe":false},"tag":"nightly-master","telemetryNamespace":"istio-system","tracer":{"datadog":{"address":"$(HOST_IP):8126"},"lightstep":{"accessToken":"","address":"","cacertPath":"","secure":true},"zipkin":{"address":""}},"trustDomain":"","useMCP":true},"grafana":{"enabled":true},"istio_cni":{"enabled":false},"kiali":{"createDemoSecret":true,"enabled":true},"mixer":{"adapters":{"stdio":{"enabled":true}},"policy":{"enabled":true,"resources":{"limits":{"cpu":"100m","memory":"100Mi"},"requests":{"cpu":"10m","memory":"100Mi"}}},"telemetry":{"enabled":true,"resources":{"limits":{"cpu":"100m","memory":"100Mi"},"requests":{"cpu":"50m","memory":"100Mi"}}}},"pilot":{"resources":{"limits":{"cpu":"100m","memory":"200Mi"},"requests":{"cpu":"10m","memory":"100Mi"}},"traceSampling":100},"sidecarInjectorWebhook":{"enableNamespacesByDefault":false,"image":"sidecar_injector","replicaCount":1,"rewriteAppHTTPProbe":false},"tracing":{"enabled":true}}
```
Comment (linsun) [inline test/demo/k8s.yaml]: is this generated?  this is a little harder to review

Comment (howardjohn) [reply test/demo/k8s.yaml]: Yes the whole k8s.yaml file is genreated with `make run-build-demo`. In istio/istio we didn't check these in, which maybe we dont need to, but it does allow users to `kubectl apply -k github.com/istio/install/demo/k8s.yaml`.

Specifically, this part which is hard to review, is the values file serialized, so you can review the values.yaml.

But I agree, we probably need a better solution

Comment (linsun) [reply test/demo/k8s.yaml]: ok, i'll ignore reviewing them.  certainly not the normal k8s yaml file i see.

**Comment by linsun** [comment]
@howardjohn should we make this demo test a required test now?

**Comment by howardjohn** [comment]
@linsun We should make all of the tests required I think. I had a PR here but its failing (and the demo one is new), so once we get all them passing we should do that. Things are moving pretty quick in this repo so it is hard to keep up with all the new tests, etc

**Pull request merged**